apps/ledger-live-desktop/tests/specs/services/dapp.spec.ts

```diff
@@ -6,110 +6,198 @@ import { Drawer } from "../../page/drawer/drawer";
+
+    // You can uncomment this to make sure visually that it ends correctly
+    // await page.waitForTimeout(10000);
+  });
+});
+
+test.describe("1inch dapp", () => {
+  test.beforeAll(async () => {
+    process.env.MOCK_REMOTE_LIVE_MANIFEST = JSON.stringify([
+      {
+        id: "dummy-live-app",
+        name: "1inch",
+        private: false,
+        url: "https://app.1inch.io/",
+        dapp: {
+          nanoApp: "1inch",
+          networks: [
+            {
+              currency: "ethereum",
+              chainID: 1,
+              nodeURL: "https://eth-dapps.api.live.ledger.com",
+            },
+            {
+              currency: "bsc",
+              chainID: 56,
+              nodeURL: "https://bsc-dataseed.binance.org/",
+            },
+            {
+              currency: "polygon",
+              chainID: 137,
+              nodeURL: "https://polygon-mainnet.g.alchemy.com/v2/oPIxZM7kXsPVVY1Sk0kOQwkoIOpSu8PE",
+            },
+            {
+              currency: "arbitrum",
+              chainID: 42161,
+              nodeURL: "https://arb1.arbitrum.io/rpc",
+            },
+            {
+              currency: "optimism",
+              chainID: 10,
+              nodeURL: "https://mainnet.optimism.io",
+            },
+          ],
+        },
+        homepageUrl: "https://1inch.io/",
+        icon: "https://cdn.live.ledger.com/icons/platform/1inch.png",
+        platforms: ["android", "desktop"],
+        apiVersion: "^2.0.0",
+        manifestVersion: "2",
+        branch: "stable",
+        categories: ["swap"],
+        currencies: ["ethereum", "bsc", "polygon", "arbitrum", "optimism"],
+        content: {
+          shortDescription: {
+            en: "Exchange crypto via a Defi/DEX aggregator on Ethereum mainnet, BSC or Polygon",
+          },
+          description: {
+            en: "Exchange crypto via a Defi/DEX aggregator on Ethereum mainnet, BSC or Polygon",
+          },
+        },
+        permissions: [],
+        domains: ["http://", "https://"],
+        visibility: "complete",
+      },
+    ]);
+  });
 
-  // Doesn't seem to wait correctly for the modal to disappear as we can still comment the line above and works
-  await modal.waitForModalToDisappear();
+  test("Dapp switch chain", async ({ page, electronApp }) => {
+    const discoverPage = new DiscoverPage(page);
+    const drawer = new Drawer(page);
+    const layout = new Layout(page);
 
-  // Improve the deviceAction mocking to return a result in the webview to test
-  // await webview.getByText("Result:");
+    await layout.goToDiscover();
+    await discoverPage.openTestApp();
+    await drawer.continue();
+    await drawer.waitForDrawerToDisappear();
 
-  // You can uncomment this to make sure visually that it ends correctly
-  // await page.waitForTimeout(10000);
+    const [, webview] = electronApp.windows();
+    await webview.getByRole("button", { name: "Connect wallet", exact: true }).click();
```

Comment:
to follow Page Object Model could you please move all locators to the corresponding class page?